crates/llm-ls/src/main.rs

```diff
@@ -532,6 +533,30 @@ impl LlmService {
 impl LanguageServer for LlmService {
     async fn initialize(&self, params: InitializeParams) -> LspResult<InitializeResult> {
         *self.workspace_folders.write().await = params.workspace_folders;
+        let position_encoding = params
+            .capabilities
+            .general
+            .and_then(|general_capabilities| {
+                general_capabilities
+                    .position_encodings
+                    .map(|encodings| {
+                        if encodings.contains(&PositionEncodingKind::UTF8) {
+                            PositionEncodingKind::UTF8
+                        } else if encodings.contains(&PositionEncodingKind::UTF16) {
+                            PositionEncodingKind::UTF16
+                        } else if encodings.contains(&PositionEncodingKind::UTF32) {
+                            PositionEncodingKind::UTF32
+                        } else {
+                            // Because UTF-16 is the only mandatory encoding that the client must support,
+                            // we will use it as the default encoding.
+                            // See: https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocuments
+                            PositionEncodingKind::UTF16
+                        }
+                    })
+            });
+
+        *self.position_encoding.write().await = position_encoding.as_ref().unwrap().clone();
```

Comment:
Actually, here `general_capabilities.position_encodings` is an array containing the different encodings supported by the client.

That's why it was interesting to choose the more convenient one to avoid unneeded conversions in the `apply_content_change` function.